### DISCUSSION TITLE
outdated: An ugly but working process to install Brownie on Windows and overcome the annoying cytoolz issue.

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#952] [General]
Outdated: Please refer to this post here https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/971

Brownie installation issues and help requests regularly appear on this board. Here's a brief description of how I got it to work, to save others an hour of searching.

**Disclaimer**: I am not even sure if we need pipx, because I finally installed eth-brownie with simple `pip install eth-brownie` see below. I just got tired and it worked. If someone could please comment on how to reduce the complexity of this whole process, it would be much appreciated. The same is true for the Visual C++ build tools installation with a whopping ~5GB it needs. I am sure, we just need a tiny script from the whole thing. If someone figures that out, please comment as well!

Anyway, here we go!

**Setting the stage.**
Download and partly install the Visual C++ build tools: https://visualstudio.microsoft.com/visual-cpp-build-tools/
Selecting only the module ticked off in the screenshot (German system) is sufficient, continue with the installation.
![image](https://user-images.githubusercontent.com/32838961/152591567-dab32304-9227-4d91-ad19-8fd3659f0501.png)

Normally I use a virtual environment for my Python projects, in this case, I would advise against it because what is written [on their website](https://eth-brownie.readthedocs.io/en/stable/install.html):
_pipx installs Brownie into a virtual environment and makes it available directly from the command line. Once installed, you will never have to activate a virtual environment prior to using Brownie._

**continue ...**

1. Open Windows Powershell as administrator.
2. Install pipx with `python -m pip install pipx`.
3. Check if pipx appears as path in the environment variables with `python -m pipx ensurepath`.
4. If the path is not automatically set by the command, please set it up yourself under environment variables in Windows. (If you don't know how to do this, please use google, it's easy)
5. Last point: The brownie installation always fails because of the cytoolz. You can fix that by installing cython.  `pip install Cython`
6. Now finally we can install Brownie with `pip install eth-brownie`
7. And test with `brownie --version`

**Comment by cromewar** [comment]
Nice @n4n0b1t3 
Yes pipx is not necessary, and also may cause troubles at importing on vscode, that's why I strongly suggest to use just pip.
I've wrote a guide with this issue [here](https://medium.com/@cromewar/the-mighty-cytools-error-at-web3-smart-contract-development-with-brownie-63335d50f230), just waiting for PR confirmation. However, part of yours are more detailed than mine, can I add a part of yours on it? (of course I'll give you credits) Have you a medium account?

**Reply by n4n0b1t3** [reply]
Yes, please copy whatever you find useful. During the MS C++ build tools installation, couldn't we just deselect everything unnecessary? I am sure this could be reduced to a few MB instead of the whole motherload. Just don't know what to choose.

**Reply by cromewar** [reply]
Not likely I did some research on that and is necessary to install all the libraries related, and thanks a lot I'll update that guide with this info :D
